src/MongoDB.Driver/MongoClientSettings.cs

```diff
@@ -1073,5 +1078,15 @@ internal ClusterKey ToClusterKey()
                 _waitQueueSize,
                 _waitQueueTimeout);
         }
+
+        private void Validate()
+        {
+            if (AllowInsecureTls && SslSettings != null && SslSettings.CheckCertificateRevocation)
+            {
+                throw new InvalidOperationException(
+                        $"{nameof(AllowInsecureTls)} cannot be true when " +
+                        $"{nameof(SslSettings)}.{nameof(_sslSettings.CheckCertificateRevocation)} is true");
+            }
```

Comment:
I'm a little confused by why this validation is needed at all, and if it is needed I wonder whether the following makes more sense:

    if (_sslSettings != null && _sslSettings.CheckCertificateRevocation && _allowInsecureTls)
    {
        throw new InvalidOperationException(
            $"{nameof(SslSettings)}.{nameof(_sslSettings.CheckCertificateRevocation)} cannot be true when " +
            $"{nameof(AllowInsecureTls)} is true.");
    }

It's almost the same but kind of in the opposite direction.

Reply:
This is probably a sign that we have contradictory settings. Which might just be for historical reasons.

I'm not sure what the proper answer is here.